client/src/main/com/sinch/sdk/models/ConversationContext.java

```diff
@@ -0,0 +1,161 @@
+package com.sinch.sdk.models;
+
+import com.sinch.sdk.core.models.ServerConfiguration;
+
+/** Execution context related to Conversation domains */
+public class ConversationContext {
+
+  private final ConversationRegion region;
+  private final String url;
+  private final String templateManagementUrl;
+
+  private ConversationContext(ConversationRegion region, String url, String templateManagementUrl) {
+    this.region = region;
```

Comment:
Contrary to SMS, this constructor is private, but I think the region should not be null. Apart if you say it's defined in a configuration file and the value will never be null.

Reply:
Right.
1. SMS constructor should be private (will create a dedicated PR related to SMS to avoid in this feature branch only)
2. A default value is currently set by config file (https://github.com/sinch/sinch-sdk-java/blob/feat/conversation-rebase-main/client/resources/config-default.properties#L21)

Ideally SMS should not set a default value from constructor: with future feature (overriding request options at endpoint call level),  this will prevent to have some kind of default config (with region set) and only some context properties defined (not region)